eng/native/configurecompiler.cmake

```diff
@@ -401,9 +401,11 @@ if (CLR_CMAKE_HOST_UNIX)
     add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-stringop-truncation>)
     add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-placement-new>)
 
-    check_cxx_compiler_flag(-faligned-new COMPILER_SUPPORTS_F_ALIGNED_NEW)
-    if (COMPILER_SUPPORTS_F_ALIGNED_NEW)
-      add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-faligned-new>)
+    if (CMAKE_CXX_COMPILER_ID)
```

Comment:
This condition was missed out previously, and failing when building `libs.native` subset with gcc (which we currently don't build in the CI and maybe we can; since it takes 30-60 seconds max)